### PR TITLE
Motor interface continuation

### PR DESCRIPTION
**Pull request by omarhogni**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -21,6 +21,7 @@ include(tfc_config)
 include(tfc_warnings)
 include(tfc_docs)
 include(tfc_cpack)
+include(tfc_systemd)
```
Comment (jbbjarnason) [inline CMakeLists.txt:24]: wasn't this in previous PR?

File: libs/motor/inc/public/tfc/motors/mock_motor.hpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+ * @brief mock motor class
+ * @details This struct is only used for testing purposes.
+ * none of the methods should be called in production.
+ * This struct is also simple and does not rely on time by
+ * default.
+ */
+
+template <typename clock_t>
+struct mock_motor {
+  explicit virtual_motor(boost::asio::io_context&) {}
```
Comment (jbbjarnason) [inline libs/motor/inc/public/tfc/motors/mock_motor.hpp:29]: discrepancy here, does it compile?

File: libs/motor/inc/public/tfc/motors/virtual_motor.hpp [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    if (!config_.nominal) {
+      return motor_error(errors::err_enum::motor_missing_speed_reference);
+    }
+    [[maybe_unused]] auto frequency = tfc::motor::impl::nominal_at_50Hz_to_frequency(config_.nominal.value(), vel);
+    logger_.info("convey running at {}", frequency);
+    running_ = true;
+    return {};
+  }
+
+  void convey(QuantityOf<mp_units::isq::velocity> auto vel,
```
Comment (jbbjarnason) [inline libs/motor/inc/public/tfc/motors/virtual_motor.hpp:77]: What should occur if convey is called twice in the same event cycle?

File: libs/motor/inc/public/tfc/motors/virtual_motor.hpp [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+
+  ~virtual_motor() {}
+
+  auto convey() -> std::error_code {
+    logger_.info("convey!");
+    running_ = true;
+    return {};
+  }
+
+  auto convey(QuantityOf<mp_units::isq::velocity> auto vel) -> std::error_code {
```
Comment (jbbjarnason) [inline libs/motor/inc/public/tfc/motors/virtual_motor.hpp:66]: maybe look into making one private `convey` function which could cover all public instances, less duplication. could use non type template parameters to activate or deactivate code

File: CMakeLists.txt (hunk 1 above)
Comment (omarhogni) [reply CMakeLists.txt]: Rebased.

File: libs/motor/inc/public/tfc/motors/mock_motor.hpp (hunk 2 above)
Comment (omarhogni) [reply libs/motor/inc/public/tfc/motors/mock_motor.hpp:29]: This idea is not included anywhere i believe. Lets revisit when there are some tests written on top of this and in usage.

File: libs/motor/inc/public/tfc/motors/virtual_motor.hpp (hunk 3 above)
Comment (omarhogni) [reply libs/motor/inc/public/tfc/motors/virtual_motor.hpp:77]: If you mean someone calls convey when the conveyor is already running?
We had discussed that the new parameters would overtake the previous invocation.

f.e.
```cpp
convey(5m)
// Conveyor will move 10meters from the point it had reached in completing the 5m transition
convey(10m) 
```

Comment (omarhogni) [reply libs/motor/inc/public/tfc/motors/virtual_motor.hpp:77]: exactly. However all of this needs more maturity and testing. Especially for these edge cases.

File: libs/motor/inc/public/tfc/motor/atv320motor.hpp [hunk 5]
```diff
@@ -0,0 +1,219 @@
+#pragma once
+#include <mp-units/chrono.h>
+
+#include <iostream>  //TODO: Remove
```
Comment (jbbjarnason) [inline libs/motor/inc/public/tfc/motor/atv320motor.hpp:4]: remove?

File: libs/stx/inc/public/tfc/utils/units_glaze_meta.hpp [hunk 6]
```diff
@@ -61,6 +95,7 @@ struct to_json_schema<mp_units::quantity<ref_t, rep_t>> {
     if (!data.has_value()) {
       data = tfc::json::schema_meta{};
     }
+    assert(unit_ascii[0] != '\0');
```
Comment (jbbjarnason) [inline libs/stx/inc/public/tfc/utils/units_glaze_meta.hpp:98]:
```suggestion
```

**Review by jbbjarnason — APPROVED** [review]

**Pull request merged**